core/slim_globals.cpp

```diff
@@ -1647,17 +1647,37 @@ void SLiM_ConfigureContext(void)
 
+// BCH 12/10/2024: Removed the type field, and changed the treatment of is_null.  We have a
+// tricky problem here, which is that is_null is now variable-length and there is no count.
+// The number of byte (uint8_t) entries in is_null depends on the number of chromosomes in
+// the full set of tree sequences, because the node metadata has to contain flags (bits) for
+// every chromosome, not just for the chromosome represented by this file.  So we deduce the
+// length of is_null from that, but it is variable and has no count associated with it in
+// the metadata here.  I think this is actually not allowed in JSON Schema, understandably.
+// The schema here provides 8 bits, so it suffices for up to an 8-chromosome model.  We can
+// have up to 256 chromosomes, so we would have up to 32 uint8_t bytes of is_null data here,
+// in that extreme case.  It is very ugly, but I think the solution might be to write out
+// a different version of the metadata schema depending on the number of bytes used.  In
+// other words, if 7 bytes of is_null data are needed (for 49-56 chromosomes), we'd write
+// out a version of the schema that specifies 7 bytes of is_null data, named is_null1 to
+// is_null7, I suppose.  This effectively puts the count into the schema itself.  Peter,
+// do you see a better option than this?  I'd really prefer not to add a count, since it's
+// completely redundant.
 const std::string gSLiM_tsk_node_metadata_schema =
-R"V0G0N({"$schema":"http://json-schema.org/schema#","additionalProperties":false,"codec":"struct","description":"SLiM schema for node metadata.","examples":[{"haplosome_type":0,"is_null":false,"slim_id":123}],"properties":{"haplosome_type":{"binaryFormat":"B","description":"The 'type' of this haplosome (0 for autosome, 1 for X, 2 for Y).","index":2,"type":"integer"},"is_null":{"binaryFormat":"?","description":"Whether this node describes a 'null' (non-existant) chromosome.","index":1,"type":"boolean"},"slim_id":{"binaryFormat":"q","description":"The 'pedigree ID' of this chromosome in SLiM.","index":0,"type":"integer"}},"required":["slim_id","is_null","haplosome_type"],"type":["object","null"]})V0G0N";
+R"V0G0N({"$schema":"http://json-schema.org/schema#","additionalProperties":false,"codec":"struct","description":"SLiM schema for node metadata.","examples":[{"slim_id":123,"is_null":0}],"properties":{"slim_id":{"binaryFormat":"q","description":"The 'pedigree ID' of the haplosomes associated with this node in SLiM.","index":0,"type":"integer"},"is_null":{"binaryFormat":"B","description":"A vector of byte (uint8_t) values, with each bit representing whether the haplosome in the corresponding chromosome is a null haplosome (1) or not (0).","index":1,"type":"integer"}},"required":["slim_id","is_null"],"type":["object","null"]})V0G0N";
```

Comment:
minor point - the description should say what order the bits come in - e.g., does the least-significant bit in the first entry correspond to the first chromosome in the list, or what?

Reply:
aha, ok; wasn't sure how verbose to get

Reply:
might as well be verbose enough to use the info!

Reply:
Done, resolving.